python/cudf/cudf/utils/filterutils.py

```diff
@@ -0,0 +1,117 @@
+# Copyright (c) 2018, NVIDIA CORPORATION.
+
+import datetime
+
+import pandas as pd
+
+
+def _apply_filter_bool_eq(val, col_stats):
+    if "true_count" in col_stats and "false_count" in col_stats:
+        if val is True:
+            if (col_stats["true_count"] == 0) or (
+                col_stats["false_count"] == col_stats["number_of_values"]
+            ):
+                return False
+        elif val is False:
+            if (col_stats["false_count"] == 0) or (
+                col_stats["true_count"] == col_stats["number_of_values"]
+            ):
+                return False
+    return True
+
+
+def _apply_filter_not_eq(val, col_stats):
+    return ("minimum" in col_stats and val < col_stats["minimum"]) or (
+        "maximum" in col_stats and val > col_stats["maximum"]
+    )
+
+
+def _apply_predicate(op, val, col_stats):
+    # Sanitize operator
+    if op not in {"=", "==", "!=", "<", "<=", ">", ">=", "in", "not in"}:
+        raise ValueError(
+            '"{0}" is not a valid operator in predicates.'.format(op)
+        )
+
+    has_min = "minimum" in col_stats
+    has_max = "maximum" in col_stats
+    has_sum = "sum" in col_stats
+    col_min = col_stats["minimum"] if has_min else None
+    col_max = col_stats["maximum"] if has_max else None
+    col_sum = col_stats["sum"] if has_sum else None
+
+    # Apply operator
+    if op == "=" or op == "==":
+        if _apply_filter_not_eq(val, col_stats):
+            return False
+        if pd.isnull(val) and not col_stats["has_null"]:
```

Comment:
Should we use `cudf.isnull` here to handle cudf Scalars?

Reply:
Should we add support for `cudf.isnull` in a separate PR and then use it here?

or 

Use both  `_is_null_host_scalar` & `pd.isnull` and add a todo to be replaced by `cudf.isnull`?

Reply:
Lets just leave as is and add a `TODO` to replace with `cudf.isnull`.

Reply:
Done